specs/wp-log-in-out-spec.js

```diff
@@ -88,12 +91,76 @@ test.describe( `[${host}] Authentication: (${screenSize}) @parallel @jetpack`, f
 		} );
 	} );
 
-	if ( dataHelper.hasAccountWithFeatures( 'passwordless' ) ) {
+	if ( dataHelper.hasAccountWithFeatures( '+2fa-sms -passwordless' ) ) {
+		test.describe( 'Can Log in on a 2fa account', function() {
+			let loginFlow, twoFALoginPage, twoFACode;
+			test.before( function( done ) {
+				loginFlow = new LoginFlow( driver, [ '+2fa-sms', '-passwordless' ] );
+				// make sure we listen for SMS before we trigger any
+				const xmppClient = listenForSMS( loginFlow.account );
+				xmppClient.once( 'online', () => {
+					// send sms now!
+					loginFlow.login();
+					twoFALoginPage = new LoginPage( driver );
+					twoFALoginPage.use2FAMethod( 'sms' );
+				} );
+				xmppClient.on( 'e2e-sms', sms => {
+					const twoFACodeMatches = sms.body.match( /^\d+/ );
+					twoFACode = twoFACodeMatches && twoFACodeMatches[0];
+					if ( twoFACode ) {
+						xmppClient.stop().then( () => done() );
+					}
+				} );
+			} );
+
+			test.it( 'Should be on the /log-in/sms page', function() {
+				return twoFALoginPage.displayed().then( function( displayed ) {
+					return driver.getCurrentUrl().then( ( urlDisplayed ) => {
+						assert( urlDisplayed.indexOf( '/log-in/sms' ) !== -1, 'The 2fa sms page is not displayed after log in' );
+					} );
+				} );
+			} );
+
+			test.it( 'Enter the 2fa code and we\'re logged in', function() {
+				return twoFALoginPage.enter2FACode( twoFACode );
```

Comment:
How do we assert that we're actually logged-in? From what I understand, and I'm fairly new to that code-base, this only current checks if the 2FA dialog disappears, but nothing more, should we maybe look for user's name or something like that to assert we're indeed correctly logged in?

Reply:
we check that we can see the reader page, this is not part of the test yet i think